flink-connectors/flink-connector-kinesis/src/main/java/org/apache/flink/streaming/connectors/kinesis/internals/ShardConsumer.java

```diff
@@ -381,13 +402,15 @@ private GetRecordsResult getRecords(String shardItr, int maxNumberOfRecords) thr
 			} catch (ExpiredIteratorException eiEx) {
 				LOG.warn("Encountered an unexpected expired iterator {} for shard {};" +
 					" refreshing the iterator ...", shardItr, subscribedShard);
-				if (SentinelSequenceNumber.getAllSentinelSequenceNumbers().contains(lastSequenceNum)) {
+				if (isSentinelSequenceNumber(lastSequenceNum)) {
```

Comment:
Now this if-else branch is basically identical to the `getInitialShardIterator` method, minus the fact that `lastSequenceNum` will never be in the middle of an aggregated record (which is fine).
I don't think the new warning log adds too much value, TBH.

So, maybe rename `getInitialShardIterator` to just `getShardIterator` and use it here.